cmd/ocitools/generate.go

```diff
@@ -0,0 +1,244 @@
+	cli.StringFlag{Name: "user", Usage: "user namespace"},
+	cli.StringFlag{Name: "uts", Usage: "uts namespace"},
+	cli.StringFlag{Name: "selinux-label", Usage: "process selinux label"},
+	cli.StringFlag{Name: "mount-label", Usage: "selinux mount context label"},
+	cli.StringSliceFlag{Name: "tmpfs", Usage: "mount tmpfs"},
+	cli.StringSliceFlag{Name: "args", Usage: "command to run in the container"},
+	cli.StringSliceFlag{Name: "env", Usage: "add environment variable"},
+	cli.StringFlag{Name: "cgroups-path", Usage: "specify the path to the cgroups"},
+	cli.StringFlag{Name: "mount-cgroups", Value: "no", Usage: "mount cgroups (rw,ro,no)"},
+	cli.StringSliceFlag{Name: "bind", Usage: "bind mount directories src:dest:(rw,ro)"},
+	cli.StringSliceFlag{Name: "prestart", Usage: "path to prestart hooks"},
+	cli.StringSliceFlag{Name: "poststart", Usage: "path to poststart hooks"},
+	cli.StringSliceFlag{Name: "poststop", Usage: "path to poststop hooks"},
+	cli.StringFlag{Name: "root-propagation", Usage: "mount propagation for root"},
+	cli.StringFlag{Name: "os", Value: runtime.GOOS, Usage: "operating system the container is created for"},
+	cli.StringFlag{Name: "arch", Value: runtime.GOARCH, Usage: "architecture the container is created for"},
+	cli.StringFlag{Name: "cwd", Value: "/", Usage: "current working directory for the process"},
+	cli.StringSliceFlag{Name: "uidmappings", Usage: "add UIDMappings e.g HostID:ContainerID:Size"},
+	cli.StringSliceFlag{Name: "gidmappings", Usage: "add GIDMappings e.g HostID:ContainerID:Size"},
+	cli.StringSliceFlag{Name: "sysctl", Usage: "add sysctl settings e.g net.ipv4.forward=1"},
+	cli.StringFlag{Name: "apparmor", Usage: "specifies the the apparmor profile for the container"},
+	cli.StringFlag{Name: "seccomp-default", Usage: "specifies the the defaultaction of Seccomp syscall restrictions"},
+	cli.StringSliceFlag{Name: "seccomp-arch", Usage: "specifies Additional architectures permitted to be used for system calls"},
+	cli.StringSliceFlag{Name: "seccomp-syscalls", Usage: "specifies Additional architectures permitted to be used for system calls, e.g Name:Action:Arg1_index/Arg1_value/Arg1_valuetwo/Arg1_op, Arg2_index/Arg2_value/Arg2_valuetwo/Arg2_op "},
+	cli.StringSliceFlag{Name: "seccomp-allow", Usage: "specifies syscalls to be added to allowed"},
+	cli.StringSliceFlag{Name: "seccomp-errno", Usage: "specifies syscalls to be added to list that returns an error"},
+	cli.StringFlag{Name: "template", Usage: "base template to use for creating the configuration"},
+	cli.StringSliceFlag{Name: "label", Usage: "add annotations to the configuration e.g. key=value"},
+}
+
+var generateCommand = cli.Command{
+	Name:   "generate",
+	Usage:  "generate a OCI spec file",
+	Flags:  generateFlags,
+	Before: before,
+	Action: func(context *cli.Context) error {
+		spec := generate.GetDefaultTemplate()
+		template := context.String("template")
+		if template != "" {
+			var err error
+			spec, err = generate.LoadTemplate(template)
```

Comment:
@mrunalp , are you suggesting move the following code into a new func `generate.NewFromTemplate`?

    template := context.String("template")
    if template != "" {
        var err error
        spec, err = generate.LoadTemplate(template)